app/main/lib/shared_models/audio_model.py

```diff
@@ -212,14 +202,15 @@ def get_context_for_search(self, task):
         return context
 
     def search(self, task):
-        audio, temporary = self.get_audio(task)
-        context = self.get_context_for_search(task)
+        body = task.get("body", {})
+        audio, temporary = self.get_audio(body)
+        context = self.get_context_for_search(body)
         if audio:
-            threshold = task.get('threshold', 0.0)
+            threshold = task.get("raw", {}).get('threshold', 0.0)
```

Comment:
looks like we are making commitments that "raw" has  `threshold` and `limit` params, I think this should be enforced somewhere

Reply:
I don't know how much we want to enforce this stuff - if we *do* want to enforce it, it's way way upstream of here, at similarity.py/model_response_package function - that's where this initially gets set.

Reply:
I agree with the general point here. We've gotten bitten multiple times by issues of parameters being passed / not passed correctly. That said, I think we should open a separate ticket and address it independently of this work. Therefore I'm going to resolve this conversation for now.